pkg/operator/controller/controller.go

```diff
@@ -38,6 +37,13 @@ const (
 	ClusterIngressFinalizer = "ingress.openshift.io/default-cluster-ingress"
 )
 
+var log = logf.Logger.WithName("cluster-ingress-controller")
+
+func init() {
```

Comment:
This seems to shift the problem around. Use case: we decide to add a new controller in a sibling package (we actually want to do this for status calc). Either we duplicate the init boilerplate in the new controller package (unnecessary and conflicting calls to `SetRuntimeLogger`), or contexts in which it would make sense to use just one of the controllers (a test) would have to import this other controller's package to enable logging (confusing).

If logging is wired at a layer above the controller-runtime consumers (e.g. controllers), say at the entrypoint, or maybe the operator package, then we can freely add new controller-runtime consumers without having to deal with package-scoped wiring.

Reply:
Ah... that makes sense. I thought I would have to call SetLogger in the controller pkg too, but I was wrong.